### ISSUE TITLE
AMP Review 2

### ISSUE THREAD

**Issue by andrewrreed** [fastforwardlabs/active-learning-cml#1]
# AMP Review 2

💥  This AMP is looking awesome! I really like the workflow you've built in the UI - it really helps the user understand the process. The repo is in a really good spot for the most part. I just have a few comments on documentation in the `README` and in the UI itself that I've outlined below. If any of these are major hurdles that cannot be implemented easily, feel free to push back and we can discuss together. 

### README.md

- [x] Minor re-wording
![image](https://user-images.githubusercontent.com/23462538/106191145-7b727300-6178-11eb-8220-251a9f497d5e.png)

- [x]  Might be good to specify which MNIST dataset (e.g. handwritten digits) since there are others. Also would be good to link to the original dataset here.
![image](https://user-images.githubusercontent.com/23462538/106191369-c7bdb300-6178-11eb-9842-37935f253167.png)

- [x]  I think this should say `apps/app.py` and might also be good to include a link to that directory. This comment also applies to the "Starting the application within CML or CDSW" section
![image](https://user-images.githubusercontent.com/23462538/106192236-e7091000-6179-11eb-8594-e078a91e49c5.png)

## Active Learner UI Feedback

- [x] I think it would be helpful to add a bit more context at the introduction of the UI to help orient the user to what they are looking at. After the opening sentence (shown below with red arrow), it might be good to include something like:

> In this application, we use the MNIST handwritten digit dataset to illustrate the active learning workflow. The workflow begins by training a convolutional neural network to recognize what digit is present in each image using only a small sample of labeled examples. We then compute the loss and accuracy of this model on a hold out validation set and are presented with ten unseen, ambiguous (as defined by the model) examples which we can manually assign labels for. Those ten examples are then added into the original training set which helps the model learn to distinguish between ambiguous classes more effectively. This process can be repeated until an acceptable level of performance is achieved. For step by step instructions, click the "Learn More" button below.

![image](https://user-images.githubusercontent.com/23462538/106192651-79a9af00-617a-11eb-8757-f67964dc97d3.png)

### The next few items correspond to the image below
![image](https://user-images.githubusercontent.com/23462538/106194678-24bb6800-617d-11eb-97be-855edf0aad1f.png)
- [x] Would it be possible to make these bullets into an ordered list (1,2,3,4) instead of bullet points?
- [x] In the third bullet, what are the embeddings that are being referenced here? Can we clarify that they are the weights from the last layer (or whatever they happen to be)?
- [ ] The last bullet says "Repeat the process until you have achieved the desired performance"... it seems there isn't currently an easy way to quickly obtain a performance metric without scanning the loss/accuracy plot for the maxima/minima. I think it would be super helpful to add a set of KPI's to the page that spells out:
   - here's the best loss/accuracy from the previous iteration
   - here's the best loss/accuracy from the current iteration
   - [maybe] here's the % change in performance by labeling 10 more datapoints

### Scatter Plot & Loss/Accuracy Plots
- [x] Can we add a title (and maybe a description) to the top of the scatterplot to help orient users to the UI? Basically just helping people understand that each dot represents an image and is spatially organized by the UMAP projection of some set of weights/embeddings
- [x] Also could we add a title to both the loss and accuracy plots specifying this is the training profile _for the most recent_ training cycle

**Comment by nishamuktewar** [comment]
- Scatter Plot & Loss/Accuracy Plots: Have added generic titles which I think make sense because otherwise it will be too wordy
- Took care of all the feedback except for the KPI thing, which would at this point be another UI feature.

**Comment by andrewrreed** [comment]
This looks good - I think the KPI's would be a good addition to the AMP in the future to help make the performance boost by retraining on newly labeled data even more apparent to the user, but for now I don't think it's make or break. Thanks!
